### PR TITLE
#1385 Integrate Sphinx & Read the Docs

### PR DESCRIPTION
**Pull request by vitojeng**
For preview this PR result: https://vitojeng-ohara.readthedocs.io/en/1385-docs/

required by #1385

### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #issue_number)
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -1,25 +1,13 @@
...
-## for developer
-
-Apart from contributing code to ohara, you can also leverage ohara to design your [custom connector](custom_connector.md),
-[custom streamapp](custom_streamapp.md) or build your UI interface via [Ohara REST interface](rest_interface.md).  
-
-1. [REST interface](rest_interface.md)
-1. [Custom Connector](custom_connector.md)
-1. [Custom StreamApp](custom_streamapp.md)
\ No newline at end of file
+Please read [this](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md) for how to contribute OharaStream.
```
Comment (chia7712) [inline docs/README.md]: Ditto. Why not keeping the direct link? The new link you add is totally not related to removed stuff.

File: docs/user_guide.md [hunk 2]
```diff
@@ -49,7 +49,7 @@ Open your browser (we recommend [Google Chrome](https://www.google.com/intl/zh-T
 ## Installation
 
 We all love docker, right? All ohara services are executed by docker container. However, it is ok to run ohara services
-through [assembly file](../README.md#build-binary) if you really really really hate docker.
+through [assembly file](https://github.com/oharastream/ohara/blob/master/README.md#build-binary) if you really really really hate docker.
```
Comment (chia7712) [inline docs/user_guide.md]: This absolute path breaks the multi version of ohara docs. Is it illegal to use relative path?

File: docs/README.md [hunk 3]
```diff
@@ -1,25 +1,13 @@
...
 
-## for user
+* [Latest](https://oharastream.readthedocs.io/en/latest/index.html) document
+* Stable document(not available now)
 
-This section is for the ohara users who are going to install and then test official streaming application. For this case,
-you don't need to compile, build or write anything for ohara. All you have to read the [User Guide](user_guide.md)
-and then follow the advice of [installation section](user_guide.md#installation).
 
-1. [User Guide](user_guide.md)
```
Comment (chia7712) [inline docs/README.md]: Why do you remove the introduction to user guide?

Comment (vitojeng) [reply docs/README.md]: The content of README.md is the home of the document, so I moved to **index.rst**
Please see [here](https://vitojeng-ohara.readthedocs.io/en/1385-docs/)

File: docs/user_guide.md (hunk 2 above)
Comment (vitojeng) [reply docs/user_guide.md]: You are right. We use ReadtheDocs for document hosting, so the relative path cannot use any more. The absolute path is a temporary solution.

**Comment by chia7712** [comment]
Could we remove “vitojeng” from the url?

File: docs/README.md (hunk 1 above)
Comment (vitojeng) [reply docs/README.md]: Yes. We can use the relative path in here. Will change it.
Thanks a lot.

**Comment by vitojeng** [comment]
> Could we remove “vitojeng” from the url?

This is just for preview because this PR not merged yet. 😆

**Comment by chia7712** [comment]
The readme in /docs is a entry point to our docs. We can just add a link to readthedocs since we are migrating to it. The introduction to each file (user_guide) can be moved to index.rst

**Comment by chia7712** [comment]
@vitojeng Do you intend to move all jpg files to ohara repo in this PR?

**Comment by vitojeng** [comment]
> @vitojeng Do you intend to move all jpg files to ohara repo in this PR?

All image files just move into **doc/src/_static/images** folder.
Thanks @konekoya for reducing the image file size. 

BTW, There are still some markdown files in the other folder, I will address this in another PR.

**Comment by chia7712** [comment]
圖片檔的路徑為何是src/_static/images? 不能只是arc/images嗎？

**Comment by vitojeng** [comment]
> 圖片檔的路徑為何是src/_static/images? 不能只是src/images嗎？

曾有試過不過沒成功, 想說先這樣... XD
不過我再多試試~

**Pull request merged**